src/libraries/Microsoft.PowerFx.Core/Types/Enums/EnumStore.cs

```diff
@@ -430,21 +433,27 @@ internal void ResetCustomEnums()
                 return RegenerateEnumTypes();
             });
 
-            foreach (var enumSpec in EnumDict)
+            return CollectionUtils.EnsureInstanceCreated(ref _enumsUsedPowerApps, () =>
             {
-                Contracts.Assert(DName.IsValidDName(enumSpec.Key));
+                var list = ImmutableList.CreateBuilder<Tuple<DName, DName, DType>>();
```

Comment:
>ImmutableList [](http://example.com/codeflow?start=27&length=13)

curious - why Immutable list instead of just List? 
Either way, we're returning an IEnumerable, so it will be readonly.

Reply:
Just to make it 100% readonly. I've seen this: ((List<string>)result).Mutate()